.github/workflows/telegram-bot.yml

```diff
@@ -37,13 +42,14 @@ jobs:
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
           # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-      - name: Test with pytest
-        run: |
-          pytest
-
+      - name: Unit test
+        run: pytest tests/unit/
+      # - name: End-to-end test
+      #   run: pytest tests/e2e/
 
   doc:
     needs: test
+    if: github.ref == 'refs/heads/master'
```

Comment:
Why don't you use two unique workflows? One for any push and one only for push to `master` branch.
```
.github/workfows
├── tests.yml
└── docs.yml
```

Reply:
`needs: test` will work anyway? or should I remove it?

Reply:
If a PR is merged into `master`, you know the `tests` have already passed :smiley:

Reply:
now, yes 😁